modules/tftp/server.rb

```diff
@@ -108,6 +108,30 @@ def pxeconfig_file(mac)
     end
   end
 
+  class Pxegrub2targetos < Server
+    def setup_bootloader(mac, os)
+      pxeconfig_dir_mac = pxeconfig_dir(mac)
+      FileUtils.mkdir_p(config_dir)
+      bootloader_directory = (Proxy::TFTP::Plugin.settings.bootloader_universe || "/usr/local/share/bootloader-universe")
+      FileUtils.cp("#{bootloader_directory}/#{os}/grubx64.efi", "#{pxeconfig_dir_mac}/grubx64.efi")
+      FileUtils.cp("#{bootloader_directory}/#{os}/shimx64.efi", "#{pxeconfig_dir_mac}/shimx64.efi")
+      File.write(File.join(pxeconfig_dir(mac), 'targetos'), os)
+    end
+
+    def pxeconfig_dir(mac = nil)
```

Comment:
Does this make sense without a mac address or should it just fail?

Reply:
The `pxeconfig_dir` without a mac address is used to copy the `grub.cfg-*` files to the generic directory